patches/server/0731-Implement-new-experience-API.patch

```diff
@@ -0,0 +1,50 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: jakebacker <jbacker42@gmail.com>
+Date: Sat, 24 Jul 2021 19:51:06 -0400
+Subject: [PATCH] Implement new experience API
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+index 9e8918d03b8213e5f6689fc93030138fd704aca9..c4286194295844b9fe31444c133b4688dac7a17c 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+@@ -1337,6 +1337,39 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
+         this.getHandle().totalExperience = exp;
+     }
+ 
++    // Paper start
++    private long getExperienceToNextLevel(int level) {
++        if (level <= 15) {
++            return 2 * level + 7;
++        } else if (level <= 30) {
++            return 5 * level - 38;
++        } else {
++            return 9 * level - 158;
++        }
++    }
++
++    private long getTotalExperienceCollected(int level) {
++        if (level <= 16) {
++            return (level * level) + (6 * level);
++        } else if (level <= 31) {
++            return (long)((2.5 * (level * level)) - (40.5 * level) + 360);
++        } else {
++            return (long)((4.5 * (level * level)) - (162.5 * level) + 2220);
++        }
```

Comment:
This is a big block fully of magic numbers and formulas. Probably add some comments directing people to where this information was gotten, and comments on the src where you gleaned this from to catch any diff that is made there.